### DISCUSSION TITLE
Issues upgrading from 4.9 to 4.19

### DISCUSSION THREAD

**Discussion by benslack19** [mmistakes/minimal-mistakes#2566] [Q&A]
## Environment

- Minimal Mistakes version: 4.9.0 trying to update to 4.19
- Ruby gem or remote theme version: (checked after `bundle update`) ruby 2.5.0p0 (2017-12-25 revision 61468) [x86_64-darwin15], rails (6.0.3.1)
- Jekyll version: 4.1.0 (checked after doing `bundle update`)
- Git repository URL: https://github.com/benslack19/benslack19.github.io
- GitHub Pages hosted (if yes provide URL to site): https://benslack19.github.io
- Operating system: MacOS High Sierrra (10.13.6)

## Expected behavior

Hello,

My current Minimal Mistakes theme is 4.9.0 and I am trying to update to the current version 4.19. (I realize this big jump may be part of my troubles.) I followed the [use-git](https://mmistakes.github.io/minimal-mistakes/docs/upgrading/#use-git) option of upgrading.

I believed I followed the steps carefully and then looked around for suggestions, including:
1. Resolving conflicting files
2. Clearing my browser cache
3. Re-checked my GitHub pages site. I did not see changes reflected and inspecting a page's element still shows it is as using 4.9.0.
4. Ran `bundle update`.
5. I also tried doing a couple of steps suggested in [this issue](https://github.com/mmistakes/minimal-mistakes/issues/2271), including `bundle exec jekyll build` and `bundle exec jekyll serve`.

This is where I feel a little stuck and I would appreciate any help.

## Steps to reproduce the behavior
N/A

## Other

This is a lower priority and I don't think it would fix the issue above, but after pushing, I got this notification:
"A file was included in `_pages/category-archive.html` that is a symlink or does not exist in your `_includes` directory." I've since put the page there but the notification still shows up.

**Comment by mmistakes** [comment]
I'd suggest going with a [fresh install of the theme](https://github.com/mmistakes/mm-github-pages-starter/generate) which installs as a `remote_theme`.

You likely have something missing or didn't fully upgrade all of the `_includes`, `_layouts`, or`_sass`. Installing as a remote theme you don't need these files in your repo since they're pulled down from the theme. You only need files in `_includes` or `_layouts` if you're overriding or customing the themes`.

After install you can then migrate your config, content (posts, pages, images, etc) and go from there.

**Comment by benslack19** [comment]
Thanks for the quick response Michael. Can you verify that these would be the steps I'd take for a fresh install?

1. Backup my current files(let's call it "backup benslack19.github.io")
2. Delete my current repo that is curently live on Github (calling it "old benslack19.github.io").
3. Delete my local repo of "old benslack19.github.io".
4. Create a fresh install of the theme via the mm-github-pages-starter (the link you sent me in your reply). The repo would be my "new benslack19.github.io" repo. Clone "new benslack19.github.io" locally.
5. Migrate my content from "backup benslack19.github.io" .

In case anything goes wrong, could I just copy things from "backup benslack19.github.io"? I had made some customizations on a few pages.

And since I'm deleting "old benslack19.github.io", I'd lose my git history for any posts, correct? It'd be a nice to keep history (to document some edits), but I'm not going to stress over it if doing the fresh install is the easiest/best approach.

**Comment by mmistakes** [comment]
I would do this all locally. Create a new branch and test everything locally first. Once that’s good you can push up a branch and create a pull request to merge it into master. That’s the benefit of using git, it version controls everything. You don’t need to remove the repo, branching is what you want.

Might be easier for you to rename _includes, _layouts, and _sass folders to something like _includes-bak, _layouts-bak, _sass-bak as well. Then if you need files from them you can move over as needed before removing completely.

**Comment by benslack19** [comment]
OK thanks, I'll give this a shot.

**Comment by benslack19** [comment]
Michael, I'm sorry but I'm still having issues. I might be doing something trivially wrong that you might catch. Here's what I did.

1. Made a new branch off the master. Renamed the folders _includes-bak, _layouts-bak, _sass-bak. (I did the same for all files that were in the _data, _pages, and _posts.

2. I downloaded the mm-github-pages-starter files from [here](https://github.com/mmistakes/mm-github-pages-starter) and put all the files into my local directory of that branch.

3. I did a `bundle exec jekyll build` to run locally but got this error (I truncated the whole message).
 
```
Liquid Exception: Syntax Error in tag 'highlight' while parsing the following markup: Valid syntax: highlight <lang> [linenos] in CHANGELOG.md
bundler: failed to load command: jekyll (/usr/local/bin/jekyll)
SyntaxError: Syntax Error in tag 'highlight' while parsing the following markup:

  

Valid syntax: highlight <lang> [linenos]

  /usr/local/lib/ruby/gems/2.5.0/gems/jekyll-3.8.7/lib/jekyll/tags/highlight.rb:21:in `initialize'
  /usr/local/lib/ruby/gems/2.5.0/gems/liquid-4.0.3/lib/liquid/tag.rb:9:in `new'
  /usr/local/lib/ruby/gems/2.5.0/gems/liquid-4.0.3/lib/liquid/tag.rb:9:in `parse'
  /usr/local/lib/ruby/gems/2.5.0/gems/liquid-4.0.3/lib/liquid/block_body.rb:34:in `parse'
  /usr/local/lib/ruby/gems/2.5.0/gems/liquid-4.0.3/lib/liquid/block.rb:58:in `parse_body'
  /usr/local/lib/ruby/gems/2.5.0/gems/liquid-4.0.3/lib/liquid/block.rb:12:in `parse'
```

I found [this string](https://github.com/mmistakes/minimal-mistakes/issues/1875) where others encountered similar issues. But I think I've taken care of most of the troubleshooting steps since I've used the original remote theme files directly and haven't edited anything.

You can see my files on this branch of my repo [here](https://github.com/benslack19/benslack19.github.io/tree/theme_upgrade2).

**Comment by iBug** [comment]
Read the error log carefully:

```
Valid syntax: highlight <lang> [linenos] in CHANGELOG.md
                                            ^^^^^^^^^^^^
```

It's something you should get rid of.

In addition, I recommend you remove anything unnecessary from your repository to avoid getting cluttered up.

**Comment by benslack19** [comment]
OK thank you, those tips helped a LOT. I've got your example site working locally now. Thank you! I'll go through the process of moving my posts, projects, and customizations now and let you know if there are any issues.

**Comment by benslack19** [comment]
Michael, things are looking good locally on my test branch. I can do some minor tweaks and cleanup but otherwise I'm ready to make it go live. I'm still getting better at using `git` so I just want to check that I'm doing the last few steps with best practices. Locally, should I go to the `master` branch merge in my test branch (`theme_upgrade2`), then push it to the remote repo?

**Comment by mmistakes** [comment]
Yes that sounds correct to me.

**Comment by benslack19** [comment]
Sorry I'm seeing another thing that I'm not sure how to fix. Do you know how to get rid of the outline around the empty space of a table? This table is produced from `pandas` output but is in a markdown file.

Below is a screenshot. On the left, is how it currently is on my `master` branch and on the right is the `theme_upgrade2` branch that I'm testing locally.

![image](https://user-images.githubusercontent.com/20425799/85206796-44caa500-b2d9-11ea-99f3-37f0e07f6cd4.png)

My preference is to have it show up like it is on the right but without the outline of the empty space. Here is what I tried based on this discussion [here](https://github.com/mmistakes/minimal-mistakes/issues/759).

1. I put the snippet below in the `/assets/css/main.css` directory.
```
.gsc-control-cse table, 
.gsc-control-cse tr, 
.gsc-control-cse td {
    border: 0;
}
```
2. That didn't work and so I tried copying the `_tables.css` file from my backup into the `/assets/css/` with that snippet.

3. I added back a `_sass` and moved the `_tables.css` file there. Still no luck. You can see that file [here](https://github.com/benslack19/benslack19.github.io/tree/theme_upgrade2/_sass).

The markdown file of the screenshot example is [here](https://github.com/benslack19/benslack19.github.io/blob/theme_upgrade2/_posts/2020-06-05-SQL-histogram-recursion.md). I appreciate any assistance.

**Comment by benslack19** [comment]
I was able to figure this out! Here's the screenshot of the update.

![image](https://user-images.githubusercontent.com/20425799/85215080-51c2b500-b328-11ea-839f-811359103d21.png)

My fix was copying the whole `_scss` directory from the full master version and then I added the line down below.

```
table {
  display: block;
  margin-bottom: 1em;
  width: 100%;
  font-family: $global-font-family;
  font-size: $type-size-6;
  border-collapse: collapse;
  border: 0;                               // eliminate border   <---------------------- edited here
  overflow-x: auto;

  & + table {
    margin-top: 1em;
  }
}
```

I think I'm good to go with the upgraded theme. Thanks!
